### ISSUE TITLE
The `ImageUploadEditing` feature should require `Clipboard`

### ISSUE THREAD

**Issue by jodator** [ckeditor/ckeditor5-image#255]
The plugin uses `Clipboard` but it does not require it. See that plugin have `Notifications` in `requires()` call.

https://github.com/ckeditor/ckeditor5-image/blob/cd2a8fd335708922efd400ad541bf09cd3d28d50/src/imageupload/imageuploadediting.js#L101

**Comment by Reinmar** [comment]
Not necessarily: https://github.com/ckeditor/ckeditor5-image/pull/248#pullrequestreview-178701770

**Comment by jodator** [comment]
But it does not makes sense for me IMHO. Also errors does not help here and you have to load it by yourself if it is not available.

For me it makes using sub plugins even harder and requires one to read the code to know what plugins to include besides this one. Shouldn't `requires()` solve it?

**Comment by jodator** [comment]
> Yes, it would fail. But we do the same thing with other feature-feature relations. This simplifies recomposing the editor with e.g. a bit different implementation of one of those features.
>
> We use Plugin#requires() for feature-util and gluefeature-subfeature relations.

Why we put some plugins to `requires()` list and some don't? Is there a rule?
Why put `Notification` and not `Clipboard`?

**Comment by Reinmar** [comment]
Where does the error appear? In tests only. In real cases, that would not happen because in 99.9% of them people use `Essentials` plugin.

By binding two plugins together we make them harder to replace, recompose. We bind features with their utils (upload -> notification). We bind parent features with sub features (bold -> boldediting, boldui). But why binding more? That's necessary, IMO, only when it's really necessary. And so far that didn't render any issues for the users.

**Comment by Reinmar** [comment]
I know it seems counterintuitive and I understand your doubts. I'm also not 100% sure about the decision myself (that's why I haven't closed this ticket). We can discuss this further F2F if you want.

**Comment by Reinmar** [comment]
Good point. String requires would help with this. Additionally, if you'd see `'Clipboard'` there you'd know that it wasn't omitted by mistake, but that someone decided to use the string version of a requirement.
